ROADMAP.md

```diff
@@ -0,0 +1,89 @@
+# Chisel Roadmap
+
+This roadmap captures the current plans of the Chisel maintainers and can be
+changed through the usual PR process.
+Not every small feature add or planned bugfix is logged here, but the high-level
+changes and planned direction that are relevant for those using and contributing
+to the project.
+Two especially notable directions on the roadmap below are:
+
+* We are migrating Chisel to rely on, exclusively, the
+  [MLIR-based FIRRTL Compiler](https://github.com/llvm/circt)
+* Accelerating improvements and changes to Chisel and the FIRRTL specification
+
+
+## 3.5.x (and Earlier)
+
+- Continue to maintain binary compatibility on these releases
+- Continue to backport bug fixes, new features if possible, documentation
+  updates, etc.
+- Deprecations are backported to ease in users migrating to next major release
+  (3.6).
+
+## 3.6 (Target 2022 November)
+
+This release is intended to provide a smooth path for the Chisel codebase to no
+longer depend on the 
+[Scala FIRRTL Compiler (SFC)](https://github.com/chipsalliance/firrtl)
+and use the [MLIR Firrtl Compiler (MFC)](https://github.com/llvm/circt) as the
+main backend. While the SFC may continue to work to consume the `.fir` produced
+by Chisel, it will not be a requirement for adding new features to the FIRRTL
+spec and the Chisel APIs. Thus, the 3.6 release will:
+
+- Deprecate `Chisel._`
+- Turn the deprecations from previous releases into Errors
+- Support integration with [MFC](https://github.com/llvm/circt) by bringing
+  `chisel-circt` in-repo
+- Stop using [SFC](https://github.com/chipsalliance/firrtl) except to test
+  deprecated features
+- Deprecate all Chisel features that are not implemented in MFC. A nonexhaustive
+  list includes:
+  - Interval Types
+  - Fixed Point Types
+  - Injecting Aspects
```

Comment:
AOP doesn't heavily depend on SFC. Maybe we can still provide this.

Reply:
Chisel has a high, initial level of complication arising from generator semantics. Injecting Aspects exacerbate this with a further level of complexity by enabling arbitrary FIRRTL to be appended after the Chisel generator runs in an order not captured in the Chisel.

This has been used as a workaround for long-missing Chisel/FIRRTL features, e.g., separation of design/verif code which is better served by an extract test code pass (which is available in CIRCT). Now that we are supporting, or have infrastructure to support, long missing features and the evolution of FIRRTL and other IRs to support user needs, we can remove the reliance on a very bad programming paradigm.

Put differently, are injecting aspects something we want? Are they a good design point we want to support in Chisel or were they patching an underlying problem in Chisel/SFC arising from missing features? If they're not the former or are the latter, then we should get rid of them. I'm soundly in the camp that they were a bad experiment and there time has come. 😅 

That said, this list is not final. If anyone is willing to put in the work to get these features into CIRCT and keep them maintained, I don't think anybody is against that!

Reply:
> Put differently, are injecting aspects something we want? Are they a good design point we want to support in Chisel or were they patching an underlying problem in Chisel/SFC arising from missing features? If they're not the former or are the latter, then we should get rid of them. I'm soundly in the camp that they were a bad experiment and there time has come.

If you can already generate assertions outside of the main chisel modules already, then maybe you can show @sequencer how to do that without aspects. If you can cover his use case, which is also what I am interested in, then we might not need injecting aspects.